src/sbd-inquisitor.c

```diff
@@ -1307,38 +1303,52 @@ int main(int argc, char **argv, char **envp)
         } else if (strcmp(argv[optind], "ping") == 0) {
             exit_status = ping_via_slots(argv[optind + 1], servants_leader);
 
-        } else if (strcmp(argv[optind], "watch") == 0) {
+        } else if (strcmp(argv[optind], "query-watchdog") == 0) {
+            exit_status = watchdog_info();
+
+        } else if (strcmp(argv[optind], "test-watchdog") == 0) {
+            exit_status = watchdog_test();
+
+        } else {
+            exit_status = -2;
+        }
+#endif
+
+        if (strcmp(argv[optind], "watch") == 0) {
+            /* sleep $(sbd $SBD_DEVICE_ARGS dump | grep -m 1 msgwait | awk '{print $4}') 2>/dev/null */
+
+#if SUPPORT_SHARED_DISK
                 if(disk_count > 0) {
                     /* If no devices are specified, its not an error to be unable to find one */
                     open_any_device(servants_leader);
                 }
+#endif
+                /* Re-calculate timeout_watchdog_warn based on any timeout_watchdog from:
+                 * SBD_WATCHDOG_TIMEOUT, -1 option or on-disk setting read with open_any_device() */
+                if (do_calculate_timeout_watchdog_warn) {
+                    timeout_watchdog_warn = calculate_timeout_watchdog_warn(timeout_watchdog);
+                }
+
+                cl_log(LOG_DEBUG, "Delay start: %s%s%s",
+                       delay_start? "yes (" : "no",
+                       delay_start? (delay > 0 ?
+                                     get_env_option("SBD_DELAY_START") :
+                                    (disk_count == 0 ? "SBD_WATCHDOG_TIMEOUT * 2" : "msgwait")) : "",
+                       delay_start? ")" : "");
 
                 if (delay_start) {
+#if SUPPORT_SHARED_DISK
                     if (delay <= 0) {
                         delay = get_first_msgwait(servants_leader);
                     }
+#endif
+                    if (disk_count == 0 && delay <= 0) {
+                        delay = 2 * timeout_watchdog;
+                    }
 
                     sleep((unsigned long) delay);
                 }
 
```

Comment:
else {
cl_log(LOG_DEBUG, "Delay start: no");
}

probably simplifies things when doing the log with values right before the sleep.